src/ccl_power.c

```diff
@@ -450,11 +451,16 @@ void ccl_cosmology_compute_linear_power(ccl_cosmology* cosmo, ccl_f2d_t *psp, in
         break;
 
       case ccl_boltzmann_class:
-        ccl_cosmology_spline_linpower_musigma(cosmo, psp, status);
+        ccl_cosmology_spline_linpower_musigma(cosmo, psp, isitgr_flag, status);
```

Comment:
This function allows us to keep the possibility to implement MG for various boltzmann transfer functions with or without re-scaling.  Depending on the cosmology used, it will compute the GR or MG observables. Also, if called with boltzman_isitgr it does no rescaling for MG calculations. I have changed now the flag name in the code to a more meaningful one as rescaled_mg_flag. It is 0 then there is rescaling and 1 otherwise.

Reply:
Okay, I see. I can see how in the ISiTGR MG case, you don't want to do the rescaling, and you're accounting for that. I'm still confused about how this works in the GR case though. These lines are within a function called 'ccl_cosmology_compute_linear_power' - I would expect this to encompass the GR case of computing the linear power spectrum as well. Can you clarify for me what would happen if this function is called in a GR cosmology? Sorry if I'm missing something obvious.

Reply:
Sure, I explain. The called function ccl_cosmology_spline_linpower_musigma, written by @beckermr, the function is able to recognize the GR case from the MG case and treat the call as such. For the GR case, the code goes to:

 else {
      // This is the normal GR case.
      for (int i=0; i<nk; i++) {
        lk[i] = log(lk[i]);
        for (int j = 0; j<na; j++) {
          //The 2D interpolation routines access the function values pk_{k_ia_j} with the following ordering:
          //pk_ij = pk[j*N_k + i]
          //with i = 0,...,N_k-1 and j = 0,...,N_a-1.
          psout_l = ccl_f2d_t_eval(psp, lk[i], aa[j], cosmo, status);
          lpk_ln[j*nk+i] = log(psout_l);
        }
      }
    }
  }

Reply:
Okay, I see - thank you!